src/index.ts

```diff
@@ -2,9 +2,12 @@ import { Users } from './services/users';
 import { Logger } from './services/logger';
 
 import type { User, ApiConfig } from './types';
+import { createIoCContainer } from './ioc';
+
+let ioc: ReturnType<typeof createIoCContainer>;
```

Comment:
Is there any reason why you don't create ioc container right here ?
You don't need to redefine it.